lib/puppet-validator.rb

```diff
@@ -73,19 +75,59 @@ def initialize(app=nil)
     erb :index
   end
 
+  get '/load/*' do
+    location = params[:splat].first.sub(/(https?:)\/\/?/, '\1//')
+    logger.info "Loading code from: #{location}"
+
+    uri = location.downcase.start_with?('http') ? URI.parse(location) : URI.parse("https://#{location}")
+
+    case uri.host
+    when 'gist.github.com'
+      code = open("#{uri}/raw").read
+    when 'pastebin.com', 'hastebin.com'
+      code = open("#{uri.scheme}://#{uri.host}/raw#{uri.path}").read
+    else
+      logger.info "Unrecognized paste service: #{uri}"
+    end
+
+    @versions = settings.puppet_versions
+    @disabled = settings.disabled_lint_checks
+    # loads lint into global namespace, but I don't see an alternative
+    @checks   = PuppetValidator::Validators::Lint.all_checks
+    @location = location
+    @code     = sanitize_code(code)
+
+    erb :index
+  end
+
+  get '/referer' do
+    redirect "/load/#{request.referer}"
```

Comment:
What should display on this redirected page? When I clicked on this link from the gist that you created, I see an error page: https://54.186.28.212/load/https://gist.github.com/

Reply:
If you click on the link on *this page*? https://gist.github.com/anonymous/c14c85ee9c1577caa43c2253e37302fa

Hmm. What browser are you using?

Reply:
Is `request.referer` not supplying the entire referred URL for some reason?

Reply:
hmm, that would be Chrome, yeah? It means that Chrome isn't sending the referer heading properly. Will look into this, thanks!

Reply:
Yes, I'm using Chrome. I also tested on Safari and having problems there as well. Let me post what I'm seeing there.

Reply:
This is Chrome:

![image](https://user-images.githubusercontent.com/4807374/33297393-a170401a-d395-11e7-812c-01ea50b508b6.png)

Reply:
Hmm, it looks like Safari failed to work when I was not initially logged in to GitHub. Then I logged in and it worked fine. Then I tried logging out to reproduce the failure, but it continued to work fine. Weird.

Reply:
Hrm. Reading the docs a little more. It seemed that the header would be sent from http(s) to https, but not https to http. There also seems to be another rule in Chrome and Firefox that sends the origin only when going https to self-signed https.

I'll have to handle that case as well, it seems. Thanks for catching it!

Reply:
I just rescued the error. It could perhaps be more descriptive, but I'm not entirely convinced that it matters at this point. The user will see the the URL didn't follow through because the field will be truncated.